src/main/java/org/ibp/api/java/impl/middleware/common/validator/LocationValidator.java

```diff
@@ -180,34 +184,80 @@ public void validateUpdate(final Integer locationId, final LocationRequestDto lo
 		this.errors = new MapBindingResult(new HashMap<>(), LocationRequestDto.class.getName());
 
 		this.validateLocation(this.errors, locationId);
-		if (locationRequestDto.getName() != null) {
-			this.validateLocationName(locationRequestDto.getName());
-		}
-
-		if (locationRequestDto.getType() != null) {
-			this.validateLocationType(locationRequestDto.getType());
-		}
-
-		if (locationRequestDto.getAbbreviation() != null) {
-			this.validateLocationAbbr(locationRequestDto.getAbbreviation());
-			this.validateLocationAbbrNotExists(locationId, locationRequestDto.getAbbreviation());
-		}
+		this.validateLocationNotEditable(locationId);
+		this.validateLocationName(locationRequestDto.getName());
+		this.validateLocationType(locationRequestDto.getType());
+		this.validateLocationAbbr(locationRequestDto.getAbbreviation());
+		this.validateLocationAbbrNotExists(locationId, locationRequestDto.getAbbreviation());
 
 		final LocationDTO locationDTO = this.locationService.getLocation(locationId);
-		final Integer countryId = locationRequestDto.getCountryId() != null ? locationRequestDto.getCountryId() : locationDTO.getCountryId();
-		final Integer provinceId = locationRequestDto.getProvinceId() != null ? locationRequestDto.getProvinceId() : locationDTO.getProvinceId();
+
+		final Integer countryId = locationRequestDto.getCountryId();
+		final Integer provinceId = locationRequestDto.getProvinceId();
 		this.validateCountryAndProvince(countryId, provinceId);
 	}
 
 	public void validateCanBeDeleted(final Integer locationId) {
 		this.errors = new MapBindingResult(new HashMap<>(), Integer.class.getName());
 
 		this.validateLocation(this.errors, locationId);
```

Comment:
@cuenyad you can make this.validateLocation to return Location or Optional<Location> and then pass the object to the other validators instead of the id to avoid querying the same object multiple times.

Reply:
done.